src/modules/mediaoverlays/MediaOverlayModule.ts

```diff
@@ -347,6 +592,25 @@ export class MediaOverlayModule implements ReaderModule {
           this.audioElement.currentTime >= this.currentAudioEnd - 0.05
         ) {
           log.log("ontimeupdate - mediaOverlaysNext()");
+
+          if (
+            this.isSegmentMode &&
+            this.currentAudioEnd ===
+              this.getAudioTimeRangeFromNode(
+                this.mediaOverlayNodesForSegment[
+                  this.mediaOverlayNodesForSegment.length - 1
+                ]
+              )?.[1]
```

Comment:
Add extra `this.mediaOverlayNodesForSegment.length > 0` just to be safe ~

```suggestion
            this.isSegmentMode &&
            this.mediaOverlayNodesForSegment.length > 0 &&
            this.currentAudioEnd ===
              this.getAudioTimeRangeFromNode(
                this.mediaOverlayNodesForSegment[
                  this.mediaOverlayNodesForSegment.length - 1
                ]
              )?.[1]
```